File: vorstand-desktop/src/lib/types/membership-fee.ts
/** Eine Beitragszahlung (1 Mitglied + 1 Jahr) inkl. Mitglieder-Daten via JOIN. */
export interface MembershipFeePayment {
  id: string;
  member_id: string;
  year: number;
  amount: string | null;
  reference_nr: string | null;
  bank_reference: string | null;
  /** "offen" oder "bezahlt" */
  status: string | null;
  paid_date: string | null;
  payment_method: string | null;
  notes: string | null;
  created_at: string | null;
  updated_at: string | null;
  // JOIN auf members
  vorname: string | null;
  nachname: string | null;
  email: string | null;
  strasse: string | null;
  plz: string | null;
  ort: string | null;
  member_status: string | null;
}

export interface MembershipFeeSummary {
  total: number;
  paid: number;
  open: number;
  total_amount: string | null;
  paid_amount: string | null;
  open_amount: string | null;
}

export interface MarkFeePaidRequest {
  paid_date?: string;
  payment_method?: string;
  notes?: string;
}

export interface MembershipFeeSettings {
  id: string | null;
  year: number;
  amount: string | null;
  gv_date: string | null;
  due_date: string | null;
  description: string | null;
}

export interface FeeSettingsUpsert {
  year: number;
  amount: string;
  gv_date?: string | null;
  due_date?: string | null;
  description?: string | null;
}

export interface GeneratePaymentsResponse {
  created: number;
  skipped: number;
  total: number;
}

File: vorstand-android/app/src/main/java/ch/fwvraura/vorstand/data/model/MembershipFee.kt
package ch.fwvraura.vorstand.data.model

import com.google.gson.annotations.SerializedName

/**
 * Eine einzelne Zahlung (= 1 Mitglied + 1 Jahr) aus membership_fee_payments,
 * angereichert mit Mitglieder-Daten via JOIN auf members.
 */
data class MembershipFeePayment(
    val id: String,
    @SerializedName("member_id") val memberId: String,
    val year: Int,
    val amount: String? = null,
    @SerializedName("reference_nr") val referenceNr: String? = null,
    @SerializedName("bank_reference") val bankReference: String? = null,
    /** "offen" oder "bezahlt" */
    val status: String? = null,
    @SerializedName("paid_date") val paidDate: String? = null,
    @SerializedName("payment_method") val paymentMethod: String? = null,
    val notes: String? = null,
    @SerializedName("created_at") val createdAt: String? = null,
    @SerializedName("updated_at") val updatedAt: String? = null,
    // Aus dem JOIN auf members
    val vorname: String? = null,
    val nachname: String? = null,
    val email: String? = null,
    val strasse: String? = null,
    val plz: String? = null,
    val ort: String? = null,
    @SerializedName("member_status") val memberStatus: String? = null
)

/** Antwort von GET /membership-fees/summary?year=YYYY */
data class MembershipFeeSummary(
    val total: Int = 0,
    val paid: Int = 0,
    val open: Int = 0,
    @SerializedName("total_amount") val totalAmount: String? = null,
    @SerializedName("paid_amount") val paidAmount: String? = null,
    @SerializedName("open_amount") val openAmount: String? = null
)

/** Body fuer PATCH /membership-fees/payments/:id/pay */
data class MarkFeePaidRequest(
    @SerializedName("paid_date") val paidDate: String? = null,
    @SerializedName("payment_method") val paymentMethod: String? = null,
    val notes: String? = null
)
